src/app/components/filters/filters.test.tsx

```diff
@@ -0,0 +1,96 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+
+enum MockedComponentTestIds {
+  StopsFilter = 'stops-filter',
+  PriceRangeFilterProps = 'price-range-filter'
+}
+
+jest.mock('./filters.css', () => {});
+
+jest.mock('../../../components/ui/card', () => ({
+  Card: ({ children }: any) => <div>{...children}</div>
+}));
+
+jest.mock('./stops-filter/stops-filter', () => ({
+  StopsFilter: ({ activeFilter, setActiveFilter }: StopsFilterProps) => {
+    return <button data-testid={MockedComponentTestIds.StopsFilter}
+      onClick={() => setActiveFilter(Stops.NonStop)}>{activeFilter}</button>
+  }
+}));
+
+jest.mock('./price-range-filter/price-range-filter', () => ({
+  PriceRangeFilter: (props: PriceRangeFilterProps) => {
+    return <button data-testid={MockedComponentTestIds.PriceRangeFilterProps}
+      onClick={() => props.setSelectedRange({ min: props.minOnScale + 5, max: props.maxOnScale })}></button>
+  }
+}))
+
+describe(Filters.name, () => {
+  let props: FiltersProps;
+  beforeEach(() => {
+    props = {
+      min: 200,
+      max: 300,
+      setFilters: jest.fn()
+    };
+
+    jest.clearAllMocks();
+  });
+
+  it('should not render PriceRange filter component if min is not defined', () => {
+    const propsWithMinUndefined = { ...props, min: undefined };
+
+    render(<Filters {...propsWithMinUndefined}></Filters>);
+
+    const priceRangeFilterComponent = screen.queryByTestId(MockedComponentTestIds.PriceRangeFilterProps);
+    expect(priceRangeFilterComponent).toBeNull();
+  });
+
+  it('should not render PriceRange filter component if max is not defined', () => {
+    const propsWithMaxUndefined = { ...props, max: undefined };
+
+    render(<Filters {...propsWithMaxUndefined}></Filters>);
+
+    const priceRangeFilterComponent = screen.queryByTestId(MockedComponentTestIds.PriceRangeFilterProps);
+    expect(priceRangeFilterComponent).toBeNull();
+  });
+
+  it('should render PriceRange filter component if min and max are defined', () => {
+    render(<Filters {...props}></Filters>);
+
+    const priceRangeFilterComponent = screen.queryByTestId(MockedComponentTestIds.PriceRangeFilterProps);
+    expect(priceRangeFilterComponent).not.toBeNull();
+  });
+
+  it('should call the set filters with just the stops value changed when Stop filter is clicked', () => {
+    const propsToSet: FiltersProps = {...props, priceRange: { min: 20, max: 100 }, stops: Stops.MultipleStops };
+    render(<Filters {...propsToSet }></Filters>);
+
+    const StopsFilterMock = screen.getByTestId(MockedComponentTestIds.StopsFilter);
+    fireEvent.click(StopsFilterMock);
+
+    const expectedCallParams: AdditionalFilters = {
+        priceRange: propsToSet.priceRange,
+        stops: Stops.NonStop
+    };
+    expect(props.setFilters).toHaveBeenCalledWith(expectedCallParams);
+  });
+
+  it('should call the set filters with just the price range value changed when price range filter is clicked', () => {
```

Comment:
test that are same excpet some values, can be made to a theory with each

Reply:
I was going to have something like 
[test1Data, test2Data].forEach() but felt like its simpler repeated.
If this was just to strike awareness of the concept, I've used it in one place